### DISCUSSION TITLE
Ideas for features

### DISCUSSION THREAD

**Discussion by Issif** [falcosecurity/falco-talon#277] [Ideas]
I'll use this issue to list the ideas of features, feel free to propose yours and I'll add them if I think they're relevant:

- [ ] OpenTelemetry
	- [x] Prometheus metrics
	- [x] Traces
- [ ] Authentication:
	- [ ] Auth with Token
	- [ ] Auth with mTLS
	- [ ] TLS (cert + ca)
- [x] `dry-run` option in the rule
- [ ] New Actionners:
	- [x] `kubernetes:delete` to delete resources other than Pods (secrets/configmaps/deployment/statefulset/replicaset/daemonset)
	- [x] `kubernetes:script` to run a shell script in a pod
	- [ ] `kubernetes:disableuser` disable a control plane user
	- [x] `kubernetes:log` get last logs from a pod
	- [ ] `kubernetes:checkpoint` create a [checkpoint](https://kubernetes.io/blog/2022/12/05/forensic-container-checkpointing-alpha/)
	- [x] `kubernetes:tshark`, `kubernetes:tcpdump`
	- [x] `kubernetes:cordon`
 	- [x] `kubernetes:drain`
	- [ ] `ansible:playbook` run an ansible playbook locally/remotely
	- [ ] `shell:command` run a shell command locally
	- [ ] `shell:script` run a shell script locally
	- [x] `aws:lambda` run an AWS Lambda function
	- [ ] `aws:cli` run an AWS CLI Command (maybe `aws:command` is a better name?)
	- [x] `calico:networkpolicy`
	- [x] `cilium:networkpolicy`
	- [ ] `webhook:template`
	- [x] `gpc:function`
- [ ] New Notifiers:
  - [x] Elasticsearch
  - [x] Loki
  - [ ] Grafana Annotations
  - [ ] Syslog
  - [x] AWS S3
  - [ ] OTLP Logs
- [ ] Improvements
  - [x] Allow to set http method and headers for the notifier `webhook`
  - [x] Deduplication of received events
  - [x] Allow to specify namespaces in the `allow` parameter of `*:networkpolicy` 
- [ ] UX:
  - [x] Add a `debug` mode: only display the received event which match, except if `debug=true`
  - [x] Check the unicity of the rule names
  - [x] Alllow to use env vars for the static configuration
  - [x] Allow sequential actions by rule
  - [x] Allow to specify multiple rule files (with merge + override)
- [ ] Docs:
  - [x] Specify required fields in the events for the actionners
- [ ] Tests:
  - [ ] Unit tests
  - [ ] Run the Units tests in GitHub Actions

**Comment by cme-incom** [comment]
It would be great to have a Slack notifier with templating for messages. My specific issue is that I want to display which cluster triggered the Falco Talon action.

**Reply by Issif** [reply]
There's already a Slack notifier, but without templating, I'll figure why it's possible.

**Reply by cme-incom** [reply]
Yea im using it :D 
The fields im mostly interested in is:
```yaml
Action: terminate pod
Rule: falco-rule-name
Pod: default/allowed-pod-xxx-yyy
Image: registry.io/image:tag
Status: success
Cluster: xxxYYY
Timestamp: YYYMMMDDDD
```
